pandas/core/indexes/base.py

```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
 
     def where(self, cond, other=None):
         """
-        Return an Index of same shape as self and whose corresponding
-        entries are from self where cond is True and otherwise are from
-        other.
+        Return a copy of self with entries replaced where cond is False.
+
+        The replacement is taken from other.
```

Comment:
I don't think this whole description is very clear. May be you can simply mention in the short summary this is a conditional. And explain what exactly does in the extended summary.

Reply:
Okay, I've copied the description from `Series.where` and `DataFrame.where`.